File: Numerical Methods/l14/zad1.cpp
#include <iostream>
#include <math.h>

int main() {



    double a = 1;
    double b = 4;

    const int n = 5;
    double z[n+1];
    double x[n+1];

    for(int k=0;k<=n;k++) {
        //std::cout << k <<std::endl;
        z[k] = cos(((2.0*k+1)/(2.0*n+2))*3.14159);
        //std::cout<<z[k]<<std::endl;
        x[k] = (z[k]*(b-a)+a+b)/2;
        std::cout<<x[k]<<std::endl;
    }

    return 0;
}

File: Numerical Methods/l14/zad1.py
import matplotlib.pyplot as plt
import math

a = 1.0
b = 4.0

n = 8
z = []
x = []
ones = []

for k in range(n+1):
    z.append(math.cos(((2.0*k+1)/(2.0*n+2))*3.14159))
    x.append((z[k]*(b-a)+a+b)/2)
    ones.append(1)
    print(x[k])

plt.plot(x,ones,'.')
plt.show()